src/wallet.js

```diff
@@ -1586,6 +1601,127 @@ const wallet = {
   getTokenIndex(token_data) {
     return token_data & TOKEN_INDEX_MASK;
   },
+
+  /**
+   * Save rewardSpendMinBlocks variable from server
+   *
+   * @param {number} rewardSpendMinBlocks
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  updateRewardLockConstant(rewardSpendMinBlocks) {
+    this._rewardSpendMinBlocks = rewardSpendMinBlocks;
+  },
+
+  /**
+   * Return the minimum blocks required to unlock reward
+   *
+   * @return {number} Minimum blocks required to unlock reward
+   *
+   * @throws {ConstantNotSet} If the weight constants are not set yet
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  getRewardLockConstant() {
+    if (this._rewardSpendMinBlocks === null) {
+      throw new ConstantNotSet('Reward block minimum blocks constant not set');
+    }
+    return this._rewardSpendMinBlocks;
+  },
+
+  /**
+   * Clear rewardSpendMinBlocks constants
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  clearRewardLockConstant() {
+    this._rewardSpendMinBlocks = null;
+  },
+
+  /**
+   * Method called when websocket connection is opened
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  websocketOpened() {
+    this.subscribeAllAddresses();
+    this.addMetricsListener();
+  },
+
+  /**
+   * Method called when websocket connection is closed
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  websocketClosed() {
+    this.removeMetricsListener();
+  },
+
+  /**
+   * Start listening dashboard ws messages from full node
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  addMetricsListener() {
+    WebSocketHandler.on('dashboard', this.handleWebsocketDashboard);
+  },
+
+  /**
+   * Stop listening dashboard ws messages from full node
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  removeMetricsListener() {
+    WebSocketHandler.removeListener('dashboard', this.handleWebsocketDashboard);
+  },
+
+  /**
+   * Method called when received dashboard ws message from full node
+   * Right now we just update the network height
+   *
+   * @param {Object} data Metrics object ws data with 'height' key
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  handleWebsocketDashboard(data) {
+    // So far we just use the height of the network and update in the variable
+    wallet.updateNetworkHeight(data.height);
+  },
+
+  /**
+   * Update network height variable
+   *
+   * @param {number} networkHeight
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  updateNetworkHeight(networkHeight) {
```

Comment:
I think we need to do an API call to get the height in `executeGenerateWallet`, otherwise the wallet won't have it set for a few moments after starting.

Reply:
It always depends on the implementation. It's like if we lose connection to the ws, when we connect again it will update the height value and update. And the update will be propagated to the wallets that are listening to it. I don't see any reason why we should add complexity to the code with one more API call. Nothing is sync in the wallet, so it's not a problem to update the height half a second later, like we do when receiving new transactions